packages/esm-patient-orders-app/src/lab-results/useLabResultsFormSchema.tsx

```diff
@@ -0,0 +1,124 @@
+import { z } from 'zod';
+import { type LabOrderConcept, useOrderConceptByUuid } from './lab-results.resource';
+
+type SchemaRecord = Record<string, z.ZodType>;
+
+/**
+ * Custom hook to generate a Zod schema for lab results form based on a lab order concept.
+ * @param labOrderConceptUuid - The UUID of the lab order concept.
+ * @returns A Zod schema object for the lab results form.
+ */
+export const useLabResultsFormSchema = (labOrderConceptUuid: string) => {
+  const { concept, isLoading: isLoadingConcept } = useOrderConceptByUuid(labOrderConceptUuid);
+
+  if (isLoadingConcept) {
+    return z.object({});
+  }
```

Comment:
Seems safer to do this then try to deal with the possible `undefined`s down the line.

```suggestion
  if (isLoadingConcept || !concept) {
    if (!concept) {
        console.warn(`Couldn't load concept ${labOrderConceptUuid}`);
    }
    return z.object({});
  }
```